File: src/main/java/pe/nanamochi/banchus/entities/commons/SubmissionStatus.java
package pe.nanamochi.banchus.entities.commons;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum SubmissionStatus {
  FAILED(0),
  SUBMITTED(1),
  BEST(2);

  private final int value;

  public static SubmissionStatus fromValue(int value) {
    for (SubmissionStatus status : SubmissionStatus.values()) {
      if (status.getValue() == value) {
        return status;
      }
    }
    throw new IllegalArgumentException("Unknown SubmissionStatus value: " + value);
  }
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/score/enums/SubmissionStatus.kt
package pe.nanamochi.banchus.score.enums

enum class SubmissionStatus(val value: Int) {
    FAILED(0),
    SUBMITTED(1),
    BEST(2);

    companion object {
        fun fromValue(value: Int): SubmissionStatus =
            entries.find { it.value == value } ?: SUBMITTED
    }
}
